packages/nx-semantic-release/src/executors/semantic-release/semantic-release.ts

```diff
@@ -47,21 +47,44 @@ export async function semanticRelease(
   };
 }
 
-const applyProjectRoot = (value: string, root: string) =>
-  value.replace('${PROJECT_DIR}', root);
-
-const resolveOptions = (
+const applyTokens = (
   options: SemanticReleaseOptions,
   context: ExecutorContext
 ) => {
-  const root = getDefaultProjectRoot(context);
+  const PROJECT_DIR = getDefaultProjectRoot(context);
+  // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+  const PROJECT_NAME = context.projectName!;
 
-  return {
-    ...options,
-    changelogFile: applyProjectRoot(options.changelogFile, root),
-    tagFormat: options.tagFormat ?? `${context.projectName}-v\${version}`,
-    packageJsonDir: options.packageJsonDir
-      ? applyProjectRoot(options.packageJsonDir, root)
-      : undefined,
+  const replaceTokens = (value: string): string => {
+    return value
+      .replace('${PROJECT_DIR}', PROJECT_DIR)
+      .replace('${PROJECT_NAME}', PROJECT_NAME);
   };
+
+  [
+    'buildTarget',
+    'changelogFile',
+    'commitMessage',
+    'packageJsonDir',
+    'tagFormat',
+  ].forEach((option) => {
+    if (options[option]) options[option] = replaceTokens(options[option]);
+  });
+
+  if (options.gitAssets && options.gitAssets.length > 0)
+    options.gitAssets = options.gitAssets.map((asset) => replaceTokens(asset));
+
+  return options;
+};
+
+const resolveOptions = (
+  options: SemanticReleaseOptions,
+  context: ExecutorContext
+) => {
+  return applyTokens(
+    {
+      ...options,
```

Comment:
Is there a reason that we are missing the default `tagFormat` that was here before:

```typescript
  return {
    tagFormat: options.tagFormat ?? `${context.projectName}-v\${version}`
  };

```
?

Reply:
The tagFormat default was moved into the schema.json since it's using a token now and doesn't need to rely template strings.  That said it may need to move again (along with all defaults).  I'll bring up that discussion point in #10 in the near future.

Reply:
Ahh, got it!